lib/ansible/modules/system/user.py

```diff
@@ -1347,7 +1347,7 @@ def create_user(self):
                 minweeks, maxweeks, warnweeks = self.get_password_defaults()
                 try:
                     lines = []
-                    for line in open(self.SHADOWFILE, 'rb').readlines():
+                    for line in open(self.SHADOWFILE, 'r').readlines():
                         fields = line.strip().split(':')
```

Comment:
This works but in general I don't really like it as it doesn't give us any control over decoding the file (trying different encodings or using different error strategies).  How about doing it like this:

``` python
from ansible.module_utils._text import to_native

[...]

for line in open(self.SHADOWFILE, 'rb'):
    line = to_native(line, errors='surrogate_or_strict')
    fields = line.strip().split(':')
[...]
```